internal/services/conditionalaccess/conditional_access_policy_resource.go

```diff
@@ -715,19 +723,27 @@ func expandConditionalAccessSessionControls(in []interface{}) *msgraph.Condition
 	cloudAppSecurity := config["cloud_app_security_policy"].(string)
 	signInFrequency := config["sign_in_frequency"].(int)
 
-	result := msgraph.ConditionalAccessSessionControls{
-		ApplicationEnforcedRestrictions: &msgraph.ApplicationEnforcedRestrictionsSessionControl{
+	var result msgraph.ConditionalAccessSessionControls
+
+	if applicationEnforcedRestrictions {
```

Comment:
Might be good to leave a comment here explaining why we don't set it.

```suggestion
	// The API doesn't accept boolean false values, we must instead omit them
	if applicationEnforcedRestrictions {
```